### PR TITLE
Add pagination to profile page

### PR DESCRIPTION
**Pull request by vgeorge**
This contributes to #319 by adding pagination to the list of teams on the profile page.

Changes:

- Add 'knex-paginate' extension module
- Refactor "Teams" table to be reused in profile and organization pages
- Update "Paginate" component to adhere to 'knex-paginate' API
- Move route `/api/my/teams` to `/src/pages/api` folder (v1 express -> next.js)
- Update Team model to return pagination (optionally)
- Add test coverage for profile page

How to give feedback:

- Open profile spec in Cypress dashboard, after tests pass, interact with pagination element
- Review API docs for team list routes

@kamicut @batpad ready for review

### CODE REVIEW

**Comment by kamicut** [comment]
@vgeorge when interacting with the pagination component, I was able to continue after page 5 to page 6, 7, 8 and so on. The message then becomes "You are not part of a team yet". This is potentially confusing. Is there a way to disable the "Next" button in the pagination component if there are no teams left in the list?

File: src/components/tables/teams.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  return (
+    <>
+      <Table
+        data-cy={`${type}-table`}
+        rows={data}
+        columns={columns}
+        emptyPlaceHolder={isLoading ? 'Loading...' : emptyMessage}
+        onRowClick={(row) => {
+          Router.push(
+            join(URL, `/team?id=${row.id}`),
```
Comment (kamicut) [inline src/components/tables/teams.js:47]: this needs `const URL = process.env.APP_URL` at the top of the file

**Review by kamicut — CHANGES_REQUESTED** [review]

File: src/components/tables/teams.js (hunk 1 above)
Comment (vgeorge) [reply src/components/tables/teams.js]: Good catch, fixed.

**Comment by vgeorge** [comment]
@kamicut thanks for the review, I added a fix for the pagination component and it should be properly disabled now.

**Review by kamicut — APPROVED** [review]

**Pull request merged**